tests/test_cases/issue_1031/issue_1031.py

```diff
@@ -0,0 +1,24 @@
+from cocotb import test, fork
+from cocotb.triggers import Lock, Timer
+
+@test()
+async def test_trigger_lock(dut):
+
+    resource = [0]
+    lock = Lock()
+
+    fork(co(resource, lock))
+    async with lock:
+        for i in range(4):
+            resource[0] += 1
+            await Timer(10, "ns")
```

Comment:
My idea was to write a simple test that verified the context management was working by checking to see if a resource was its expected value after those points. Can you write an example of how this would work if the context management was handled from within the for loops?

Reply:
This is actually fine. You have lock acquisition, you are forcing acquisition order so you aren't depending upon scheduling behavior, and you have statements that cause scheduler re-entry while retaining the lock. Consider this resolved.